src/util/apparmor.c

```diff
@@ -57,3 +69,489 @@ int bus_apparmor_is_enabled(bool *enabledp) {
         *enabledp = enabled;
         return 0;
 }
+
+/**
+ * bus_apparmor_dbus_supported() - checks if Kernel has AppArmor support for DBus
+ * @supported:            return argument telling if AppArmor DBus is supported
+ *
+ * If the AppArmor module is not loaded, or AppArmor does not support DBus,
+ * set @supportedp to 'false', otherwise set it to 'true'.
+ *
+ * Returns: 0 if check succeeded, or negative error code on failure.
+ */
+int bus_apparmor_dbus_supported(bool *supportedp) {
+        _c_cleanup_(c_fclosep) FILE *f = NULL;
+        char buffer[LINE_MAX] = {};
+        bool supported;
+
+        f = fopen("/sys/kernel/security/apparmor/features/dbus/mask", "re");
+        if (f) {
+                errno = 0;
+                if (!fgets(buffer, sizeof(buffer), f)) {
+                        if (ferror(f) && errno != EINVAL)
```

Comment:
I assume this EINVAL check was copy-pasted. Looks like an error. I dropped it. If it was intentional, let me know! I also dropped it from the old `is_enabled()` code.